samples/README.md

```diff
@@ -33,3 +33,30 @@ Useful parameter values:
 ## Components source
 
 All samples use pre-built components. The command to run for each container is built into the pipeline file.
+
+## Sample conventions
+For better readability and functions of sample test, samples are encouraged to adopt the following conventions.
+
+* The sample should be either `*.py` or `*.ipynb`, and its file name is in consistence with its dir name.
+* For `*.py` sample, it's recommended to have a main invoking `kfp.compiler.Compiler().compile()` to compile the 
+pipeline function into pipeline yaml spec.
+* For `*.ipynb` sample, parameters (e.g., experiment name and project name) should be defined in a dedicated cell and 
+tagged as parameter. Detailed guideline is [here](https://github.com/nteract/papermill). Also, custom environment setup
+should be done within the notebook itself by `!pip install`
+
+
+## (Optional) Add sample test
+For those samples that cover critical functions of KFP, possibly it should be picked up by KFP's sample test
+so that it won't be broken by accidental PR merge. Contributors can do that by following these steps.
+
+* Place the sample under the core sample directory `kubeflow/pipelines/samples/core`
+* Make sure it follows the sample conventions.
```

Comment:
Add the link to jump to the 'sample convention' section?

Reply:
Done.

Reply:
Why not the relative link: https://support.typora.io/Links/?
markdown supports that.

Reply:
Thanks. Just learnt :p